lib/secret-tasks.js

```diff
@@ -0,0 +1,253 @@
+			};
+
+			return store.putSecret(opts)
+				.then(function(){
+					logger.info('Secret `'+opts.name+'` successfully put in store');
+				});
+		});
+};
+
+/**
+ * Delete secret from credential store
+ * @param {object} options - user provided options
+ * @param {object} options.DeploymentCenter.secret - Secret Center options
+ * @param {function} options.DeploymentCenter.secret._prefixKey - Secret key prefixer
+ * @param {object} options.secretParams - Secret parameters
+ * @param {string} options.secretParams.secret - Secret to retrieve from store
+ * @param {string} options.secretParams.version - (Optional) specify which version you want to retrieve. Will automatically handle padding of 0's
+ * @return {Q.Promise}
+ */
+exports.deleteSecret = function(options){
+	return Q.when(options)
+		.then(function(){
+			var secrets = options.DeploymentCenter.secret;
+			var params = options.secretParams;
+			var store = secrets.secretStore();
+
+			var opts = {
+				name: secrets._prefixKey(params.secret),
+				version: _.padStart(params.version, 19, '0'),
+			};
+
+			return store.deleteSecret(opts)
+				.then(function(){
+					logger.info('Secret `'+opts.name+'` (ver: '+Number.parseInt(opts.version)+') successfully deleted');
+				})
+		});
+};
+
+/**
+ * Delete secret from credential store
+ * @param {object} options - user provided options
+ * @return {Q.Promise}
+ */
+exports.listSecrets = function(options){
+	return Q.when(options)
+		.then(function(){
+			var store = options.DeploymentCenter.secret.secretStore();
+
+			return store.listSecrets()
+				.then(function(list){
+					if(list.length === 0)
+						logger.warn('There are no secrets you can see.');
+
+					for(var secret in list)
+						logger.info(list[secret]);
+
+					return list;
+				})
+		});
+};
+
+/**
+ * Verify that the secret center is configured
+ * @param {object} options - User provided options
+ * @param {object} options.DeploymentCenter - Deployment center configuration
+ * @throws Error
+ * @return options
+ */
+exports.verify = function(options){
+	return Q.when(options)
+		.then(function(){
+			if(!('secret' in options.DeploymentCenter))
+				throw new Error('Cannot manage secrets without Deployment Center `secret` configuration specified in environment');
+
+			return options;
+		})
+};
+
+/**
+ * Sets up secret center parameters into options
+ * @param {object} options - User provided options
+ * @param {string} options.AppName - Application name and secret environment
+ * @param {string} options.environment - Application environment
+ * @param {string} options.aws.region - Application region
+ * @param {object} options.DeploymentCenter.secret - Secret center configuration
+ * @return options
+ */
+exports.setupSecretCenter = function(options){
+	return Q.when(options)
+		.then(function(){
+			var app = options.AppName;
+			var env = options.environment;
+			var region = options.aws.region;
+
+			options.DeploymentCenter.secret._prefixKey = function(name) {
```

Comment:
Would it make sense to surround this with null checks, or is it guaranteed to be defined by this point?

Reply:
Looks like it is only guaranteed to be defined if `verify`  is called first, which only happens in the `create-secret-center` and `secret` commands, not in `deploy` or the other non-secret-specific commands.  Yadda needs to continue to work if secrets aren't defined.

Reply:
Good catch again.